helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java

```diff
@@ -257,11 +258,16 @@ public static boolean hasErrorPartitions(HelixDataAccessor dataAccessor, String
               continue;
             }
 
-            // We are checking sibling partition healthy status. So if partition health does not
-            // exist or it is not healthy. We should mark this partition is unhealthy.
+            // If we failed to get partition assignment for one sibling instance, we add the
+            // nstance name in return error for debugability
             if (!globalPartitionHealthStatus.containsKey(siblingInstance)
-                || !globalPartitionHealthStatus.get(siblingInstance).containsKey(partition)
+                || globalPartitionHealthStatus.get(siblingInstance).isEmpty()) {
+              unhealthyPartitions.computeIfAbsent(partition, list -> new ArrayList<>())
+                  .add(HOST_NO_STATE_ERROR + siblingInstance);
+            } else if (!globalPartitionHealthStatus.get(siblingInstance).containsKey(partition)
                 || !globalPartitionHealthStatus.get(siblingInstance).get(partition)) {
```

Comment:
could be cleaner with `globalPartitionHealthStatus.get(siblingInstance).getOrDefault()`